src/lib/cimbar_js/cimbar_recv_js.cpp

```diff
@@ -180,6 +182,20 @@ int cimbard_get_filesize(uint32_t id)
 	return md.file_size();
 }
 
+int cimbard_get_filename(const uchar* finbuffer, unsigned size, char* filename, unsigned fnsize)
+{
+	std::string fn = cimbar::zstd_header_check::get_filename(finbuffer, size);
+	if (!fn.empty())
+		fn = std::filesystem::path(fn).filename().string(); // strip path
```

Comment:
This is the basename calculation. Might hide it in a util function, just because the syntax is so mysterious.